Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx

```diff
@@ -14,48 +14,124 @@ import { useUrls } from './use-urls';
 interface IAuthContext {
     state: {
         user: IUserType;
+        loginOrGetAuthInitiated: boolean;
+        isGetAuthInfoUnauthorized: boolean;
+        hasCompletedGetAuthInfo: boolean;
+        isLoggedIn: boolean;
         loginErrorMessage: string;
     };
     actions: {
         signIn: () => void;
         signOut: () => Promise<void>;
         getUser: () => IUserType;
+        loadAuthInfo: () => Promise<void>;
         setUsername: (value: string) => void;
         setPassword: (value: string) => void;
     };
 }
 
+const defaultState = {
+    user: {
+        id: '',
+        username: '',
+        email: '',
+        permissions: { canAccessAdministration: false } as IUserPermissionsType,
+    },
+};
+
 const AuthContext = createContext<IAuthContext>({} as IAuthContext);
 
-interface IAuthProviderProps extends IHaveChildrenProps {
-    user: IUserType;
-}
+type IAuthProviderProps = IHaveChildrenProps
 
 interface ILoginDetailsType {
     Username: string;
     Password?: string;
     RememberMe: boolean;
 }
 
-const AuthProvider = ({ user, children }: IAuthProviderProps) => {
+const AuthProvider = ({ children }: IAuthProviderProps) => {
     const { startLoading, stopLoading } = useLoading();
-    const [ internalUser, setInternalUser ] = useState(user);
-    const [ username, setUsername ] = useState<string>(user.username);
+    const [ internalUser, setInternalUser ] = useState<IUserType>(defaultState.user);
+    const [ username, setUsername ] = useState<string>('');
     const [ password, setPassword ] = useState<string>();
     const [ loginErrorMessage, setLoginErrorMessage ] = useState<string>('');
     const { showError } = useNotifications();
     const defaultLoginErrorMessage = useMemo(() => 'Invalid username or password', []);
 
-    const { getLogoutUrl, getLoginSubmitUrl } = useUrls();
+    const { getLogoutUrl, getLoginSubmitUrl, getUserAuthInfoUrl } = useUrls();
 
     const {
         post: loginSubmit,
         response: loginSubmitResponse,
         status: loginSubmitStatus,
     } = useHttp<null, string, ILoginDetailsType>({ url: getLoginSubmitUrl });
 
+    const {
+        get: getAuthInfo,
+        data: authInfo,
+        isSuccess: isGetAuthInfoSuccess,
+        status: getAuthInfoStatus,
+    } = useHttp<null, IUserResponseType, null>({ url: getUserAuthInfoUrl });
+
     const { post: logout } = useHttp({ url: getLogoutUrl });
 
+    const loginOrGetAuthInitiated = useMemo(
+        () => getAuthInfoStatus !== HttpStatus.NotStarted ||
+        loginSubmitStatus !== HttpStatus.NotStarted,
+        [ getAuthInfoStatus, loginSubmitStatus ],
+    );
+
+    const hasCompletedGetAuthInfo = useMemo(
+        () => getAuthInfoStatus !== HttpStatus.NotStarted &&
+            getAuthInfoStatus !== HttpStatus.Pending,
+        [ getAuthInfoStatus ],
+    );
+
+    const isGetAuthInfoUnauthorized = useMemo(
+        () => getAuthInfoStatus === HttpStatus.Unauthorized,
```

Comment:
This call could be allowed for all users and never be Unauthorized. If I'm not logged in - the result could be `null`, otherwise my info will be returned (not anyone else's). This way we will spare handling of Unauthorized and simplify the logic.
What do you think?